docs: add missing Ory Console tabs for social sign-in providers (#1204)
ory/docs: docs/kratos/social-signin/65_netid.mdx

sidebar_label: NetID
---

```mdx-code-block
import JsonnetWarning from "../../_common/jsonnetwarning.mdx"
```

# NetID

````mdx-code-block
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
<TabItem value="console" label="Ory Console" default>

Follow these steps to add NetID as a social sign-in provider to your project using the Ory Console:

1. Sign in to Ory Console and select [**Social Sign-in**](https://console.ory.sh/projects/current/social-signin).
2. Click the switch next to the NetID logo to start the configuration.
3. Copy the **Redirect URI** and save it for later use.
4. Create a [NetID Service](https://developerzone.netid.dev/latest/devportal/tutorial/services/) and
   [NetID Client](https://developerzone.netid.dev/latest/devportal/tutorial/clients/).

> 5. While creating the NetID Client, enter the saved **Redirect URI** from Ory to the **Callback URL** field of the NetID Client.

Reviewer: ```suggestion
5. Enter the Redirect URI you copied from Ory ine the **Callback URL** field of the NetID Client.
```

Please use structure like this. Simpler, shorter.

Author: The reason for the "While" clause is that there is a bit of a contradiction in sequencing here: if you have already finished creating the NetID Client, it's already too late to fill in the **Callback URL** field. But I can go with the simpler sentence structure -- it's not too hard for the reader to figure out what to do.

Reviewer: ok

6. After creating the NetID Client, the client is initially in sandbox mode, where only specified test user email addresses can use OAuth login.

> To enable an email address for OAuth, go to the NetID Service page and click **Add test user**.

Reviewer: not clear

Author: Refactored the text to explain it a bit more:

6. After creating the NetID Client, the client is initially in _sandbox mode_, which is a pre-production state
   prior to the release of your NetID Service.
   If you want to test your NetID Client while it is in sandbox mode, you need to create one or more test users,
   by going to the NetID Service page and clicking **Add test user**.

Reviewer: To much "if's" here. Instead of describing this order of things it'd be better to add this information to an admonition and say that the client is first in sandbox, then gets promoted to production. 

Also, this information should be presented in the CLI-focused flow too, shouldn't it? If I go straight to the CLI tab I won't know this. 

Formatting: don't use italics, please. See the guidelines for formatting.

# NetID

````mdx-code-block
import JsonnetWarning from "../../_common/jsonnetwarning.mdx"
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs>
<TabItem value="console" label="Ory Console" default>

Follow these steps to add NetID as a social sign-in provider to your project using the Ory Console:

1. Sign in to Ory Console and select [**Social Sign-in**](https://console.ory.sh/projects/current/social-signin).
2. Click the switch next to the NetID logo to start the configuration.
3. Copy the Redirect URI and save it for later use.
4. Create a [NetID Service](https://developerzone.netid.dev/latest/devportal/tutorial/services/) and
   [NetID Client](https://developerzone.netid.dev/latest/devportal/tutorial/clients/).
5. Enter the Redirect URI you copied from Ory to the **Callback URL** field of the NetID Client.
6. After creating the NetID Client, the client is initially in _sandbox mode_, which is a pre-production state
   prior to the release of your NetID Service.
   If you want to test your NetID Client while it is in sandbox mode, you need to create one or more test users,

>    by going to the NetID Service page and clicking **Add test user**.

Author: I only counted one "if"! Removed italics; Added this info to the CLI section as well.

No time to do a rewrite before half past! The text is good enough as it is, I think.

7. Copy the following data from the NetID Client settings to the corresponding fields in the form on the Ory Console:
    - **Client ID**
    - **Client secret** (sandbox or live, depending on the status of the NetID Service)

> 8. In the **Scopes** field of the form on the Ory Console, add the following scopes:

Author: Updated all instances. Nothing turns up in grep.

    - `openid`
    - `email`

> 9. In the **Data Mapping** field of the form on the Ory Console, add the following Jsonnet code snippet:

Reviewer: I commented on this already but it seems you didn't get me. 
In the docs you edited, please provide an explanation of what the Jsonnet does.

You can use a simple sentence like the one that's present in the CLI variant in many cases and say where it must be added in the Console:

> Create a Jsonnet code snippet to map the desired claims to the Ory Identity schema.

Author: Added the text ", which maps the desired claims to the Ory Identity schema"
